lib/fog/vsphere/requests/compute/list_resource_pools.rb

```diff
@@ -15,20 +15,45 @@ def list_resource_pools(filters = { })
 
         # root ResourcePool + Children if they exists
         def list_raw_resource_pools(cluster)
-          [cluster.resourcePool, cluster.resourcePool.resourcePool].flatten
+          pools = []
+          traverse_raw_resource_pools(pools,cluster.resourcePool)
+          pools.uniq
         end
 
-        def resource_pool_attributes resource_pool, cluster, datacenter
+        def traverse_raw_resource_pools(pools, rp)
+          if rp != nil
+
+            if rp.respond_to? :resourcePool
+              traverse_raw_resource_pools(pools, rp.resourcePool)
+            end
+            if rp.respond_to? :each
+               rp.each do |resourcePool|
+                 traverse_raw_resource_pools(pools, resourcePool)
+               end
+            else
+              pools << rp
+            end
+          end
+        end
+
+        def resource_pool_attributes (resource_pool, cluster, datacenter)
+          name = folder_path(resource_pool).gsub(/^.*Resources(\/|)/,'')
+          if name == ''
```

Comment:
I'd change this (and the next two lines) to just be `name = 'Resources' if name.empty?`